proto/rpc.proto

```diff
@@ -28,104 +24,165 @@ message Transaction {
 }
 
 message Block {
-  uint32 version = 1;
-  Hash milestoneBlockHash = 2;
-  Hash prevBlockHash = 3;
-  Hash tipBlockHash = 4;
-  uint32 diffTarget = 5;
-  uint32 nonce = 6;
-  uint64 time = 7;
-  repeated uint32 proof = 8;
-  repeated Transaction transactions = 9;
+  string hash = 1;
+  uint32 version = 2;
+  string msHash = 3;
+  string prevHash = 4;
+  string tipHash = 5;
+  uint32 diffTarget = 6;
+  uint32 nonce = 7;
+  uint64 time = 8;
+  repeated uint32 proof = 9;
+  repeated Transaction transactions = 10;
+}
+
+message Vertex {
+    Block block = 1;
+    uint64 height = 2;
+    bool isMilestone = 3;
+    uint32 redemptionStatus = 4;
+    repeated bool txStatus = 5;
+    uint64 rewards = 6;
+}
+
+message Chain {
+    string headHash = 1;
+    uint64 DAGHeight = 2;
+    uint64 PCHeight = 3;
+    uint32 time = 4;
+}
+
+message MsChain {
+    Chain chain = 1;
+    string chainwork = 2;
+    uint64 blkDiff = 3;
+    uint64 msDiff = 4;
 }
 
 service BasicBlockExplorerRPC {
     rpc GetBlock (GetBlockRequest) returns (GetBlockResponse);
     rpc GetNewMilestoneSince (GetNewMilestoneSinceRequest) returns (GetNewMilestoneSinceResponse);
-    rpc GetLatestMilestone (GetLatestMilestoneRequest) returns (GetLatestMilestoneResponse);
+    rpc GetLatestMilestone (EmptyRequest) returns (GetLatestMilestoneResponse);
     rpc GetLevelSet (GetLevelSetRequest) returns (GetLevelSetResponse);
     rpc GetLevelSetSize (GetLevelSetSizeRequest) returns (GetLevelSetSizeResponse);
+    rpc GetVertex (GetVertexRequest) returns (GetVertexResponse);
+    rpc GetForks (EmptyRequest) returns (GetForksResponse);
+    rpc GetPeerChains (EmptyRequest) returns (GetPeerChainsResponse);
+    rpc GetRecentStat (EmptyRequest) returns (GetRecentStatResponse);
+    rpc Statistic (EmptyRequest) returns (StatisticResponse);
```

Comment:
How to determine the response with the same EmptyRequest

Reply:
by different function calls